cmd/spire-server/cli/validate/validate.go

```diff
@@ -1,42 +1,124 @@
 package validate
 
 import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+
 	"github.com/mitchellh/cli"
 	"github.com/spiffe/spire/cmd/spire-server/cli/run"
 	commoncli "github.com/spiffe/spire/pkg/common/cli"
+	"github.com/spiffe/spire/pkg/common/cliprinter"
+	"github.com/spiffe/spire/pkg/common/log"
+	"github.com/spiffe/spire/pkg/server"
 )
 
 const commandName = "validate"
 
-func NewValidateCommand() cli.Command {
-	return newValidateCommand(commoncli.DefaultEnv)
+func NewValidateCommand(ctx context.Context, logOptions []log.Option) cli.Command {
+	return newValidateCommand(ctx, commoncli.DefaultEnv, logOptions)
 }
 
-func newValidateCommand(env *commoncli.Env) *validateCommand {
+func newValidateCommand(ctx context.Context, env *commoncli.Env, logOptions []log.Option) *validateCommand {
 	return &validateCommand{
-		env: env,
+		ctx:        ctx,
+		env:        env,
+		logOptions: logOptions,
 	}
 }
 
 type validateCommand struct {
-	env *commoncli.Env
+	ctx        context.Context
+	logOptions []log.Option
+	env        *commoncli.Env
+	printer    cliprinter.Printer
 }
 
 // Help prints the server cmd usage
 func (c *validateCommand) Help() string {
-	return run.Help(commandName, c.env.Stderr)
+	return run.Help(commandName, c.env.Stderr, c.SetupPrinter)
 }
 
 func (c *validateCommand) Synopsis() string {
 	return "Validates a SPIRE server configuration file"
 }
 
+func (c *validateCommand) SetupPrinter(flags *flag.FlagSet) {
+	cliprinter.AppendFlagWithCustomPretty(&c.printer, flags, c.env, c.prettyPrintValidate)
+}
+
 func (c *validateCommand) Run(args []string) int {
-	if _, err := run.LoadConfig(commandName, args, nil, c.env.Stderr, false); err != nil {
-		// Ignore error since a failure to write to stderr cannot very well be reported
-		_ = c.env.ErrPrintf("SPIRE server configuration file is invalid: %v\n", err)
+	config, err := run.LoadConfig(commandName, args, c.logOptions, c.env.Stderr, false, c.SetupPrinter)
+	if err != nil {
+		_, _ = fmt.Fprintln(c.env.Stderr, err)
+		return 1
+	}
+	config.ValidateOnly = true
```

Comment:
It seems that we haven't test coverage from this point. We should exercise the validation of plugin configurations in unit tests.

Reply:
>The entire validate command has always used the run command

My understanding is that the current implementation of the validate command relies on run.LoadConfig, not on the run command or Server.run function, which is a significant change.
I think that the current implementation only covers the case of using bad flags in the command, and does not really exercise the result of a validation. I think that we should cover that.
Something like this (we could also check for expected values for s.stderr, etc. and make it table driven):
```go
func (s *ValidateSuite) TestValidate() {
	code := s.cmd.Run([]string{"-config", "path-to-config"})
	s.Equal(0, code, "exit code")

	code = s.cmd.Run([]string{"-config", "path-to-invalid-config"})
	s.Equal(1, code, "exit code")
}
```

Reply:
I've implemented the checks you have requested.  They use configuration files under test/fixture/config.  The failure case specifically attempts to use a disk keymanager, where the keystore_file is not specified (so it is completely within the plugin config).